Data Structures/BinaryTree/BT_CountnonLeafNodes.java

```diff
@@ -0,0 +1,119 @@
+package Git;
+
+import java.util.Scanner;
+
+public class BT_CountnonLeafNodes {
+
+	static Scanner scn = new Scanner(System.in);

```

Comment:
You never close the Scanner object. It is necessary to close the scanner object. Later on I will tell you why it is not necessary to add this object at all.

Reply:
I am generating tree by scn at various places 
Yeah I know its not necessary to pass.... scn object 
but then the user will have to pass Scanner in the construct call and he/she might have to right whole string at the console so that the program would read....
I have just created this because of the flexibility of the user who is using this program.and create own tree step by step